docs/DeveloperGuide.md

```diff
@@ -1,38 +1,269 @@
-# Developer Guide
+<div align="center">
+<h2> MoneyMind Developer Guide</h2>
+<h3> "Mind your Money" </h3>
+</div>
+
+<div align="left">
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+<img src="images\ArchitectureDiagram.png" width="300" />
+
+The ***Architecture Diagram*** given above explains the high-level design of the App. 
+
+Given below is a quick overview of how each component interact with each other.
+
+The main components in the architecture are:
+
+* `MoneyMind`: The main program of the application, 
+it initializes the other components in the correct sequence and is responsible for shut down the application.
+* `UI`: The user interface of the application.
+* `Storage`: The storage of data of the application.
+* `Data`: The data classes used in the application, including Event, Category and CategoryList.
+* `Command`: The command of the application, including AddCommand, DeleteCommand, ListCommand, etc.
+
+<img src="images\ArchitectureSequence.png" width="500" />
+
+The *sequence diagram* above shows how the components interact with each other for the scenario 
+where the user issues the command `event buy lunch e/10`.
+
+The following sections will explain the architecture in more detail.
+
+### Storage component
+
+**API**: `Storage.java`
+
+<img src="images\Storage.png" width="500">
```

Comment:
I don't know if the earlier led to this class diagram being sized wrongly, thus it is a little hard to read. (need to zoom in)
![image](https://user-images.githubusercontent.com/88264296/228425006-0d728422-af23-4116-b67d-3625439092f3.png)

Also, the attributes should be `+`, `-`, `#` and `~` instead of the icons PlantUML sets for you. (my group has also fallen into this trap)